### PR TITLE
[Feat] `useFeeProxy`

### PR DESCRIPTION
**Pull request by aidan-starke**
## Summary

- [TRN-187](https://centralitydev.atlassian.net/browse/TRN-187)
- Add `callWithFeePreferences` example

## Checklist

- [x] Add description
- [x] Tag related issue(s)

### CODE REVIEW

File: examples/substrate/use-fee-proxy/src/callWithFeePreferences.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  const api = await getChainApi("porcini");
+  const caller = createKeyring(env.CALLER_PRIVATE_KEY);
+
+  const call = api.tx.system.remarkWithEvent("Hello World");
+  const paymentInfo = await call.paymentInfo(caller.address);
+  const estimatedFee = paymentInfo.partialFee.toString();
+
+  const { paymentAsset } = argv as unknown as { paymentAsset: number };
+
+  const {
```
Comment (ken-centrality) [inline examples/substrate/use-fee-proxy/src/callWithFeePreferences.ts:29]:
```suggestion
  // querying the dex for swap price, to determine the `maxPayment` you are willing to pay
  const {
```

File: examples/substrate/use-fee-proxy/src/callWithFeePreferences.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    Ok: [_, amountOut],
+  } = (
+    await // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    (api.rpc as any).dex.getAmountsOut(estimatedFee, [XrpAssetId, paymentAsset])
```
Comment (KarishmaBothara) [inline examples/substrate/use-fee-proxy/src/callWithFeePreferences.ts:34]: Here, I think we need to query getAmountsIn

Comment (KarishmaBothara) [reply examples/substrate/use-fee-proxy/src/callWithFeePreferences.ts:34]: https://github.com/futureversecom/trn-seed/pull/636/files#diff-ad92fb37dd4e9578962efff49752afd0b028993fc0a026ee6df202bdd7db19e9R607
have a test here

Comment (aidan-starke) [reply examples/substrate/use-fee-proxy/src/callWithFeePreferences.ts]: thanks!

**Review by ken-centrality — APPROVED** [review]

**Pull request merged**